pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -183,11 +187,14 @@ protected String getQuery(SlingHttpServletRequest request) {
     protected Map<String, Object> resourceToMap(Resource resource) {
         Module module = resource.adaptTo(Module.class);
 
-        String variantName = module.getWorkspace()
+        String variantName = DEFAULT_VARIANT_NAME;
+        Stream<ModuleVariantDefinition> mvd = module.getWorkspace()
                 .moduleVariantDefinitions().get()
-                .getVariants()
-                .findFirst().get()
-                .getName();
+                .getVariants();
+
+        if (mvd != null) {
```

Comment:
instead of this conditional you coul use the traversal api. It would yield cleaner, more concise code.

Reply:
updated.

Reply:
:+1: 
(you could keep traversing all the way down to the variant name and avoid that null check :smiley: )